libmcount/plthook.c

```diff
@@ -456,6 +456,9 @@ static int setup_mod_plthook_data(struct dl_phdr_info *info, size_t sz, void *ar
 		"ld-linux-*.so.*",
 		"libdl.so.2",
 		"libdl-2.*.so",
+		/* additional misc libraries */
+		"libSegFault.so",
+		"libunwind*",
```

Comment:
I'm not sure it's worth adding them.  Did you see a problem from those?

Reply:
I can say so, but they might be unsafe for our use cases. I would like to avoid the possible problems from those libraries. Moreover, I don't think that it's useful to see the functions inside them in most cases.

Reply:
They do segmentation fault handling and backtrace handling, but those behaviors may have some conflicts to uftrace.

Reply:
I agree that they are not the target of trace most cases.  But the problem is that once it's added here there's no way to trace them.  On some rare case, user might want to trace them and that's the power of uftrace IMO.

If a function in those libraries is called from the uftrace/libmcount, hopefully the recursion guard will protect us. :)

Reply:
Okay. I will drop this too if you think so. I just wanted to reduce some possibility when tracing chrome on arm. It has a lot of problems so wanted to narrow down the problems.

Maybe the correct solution for this is to provide a way to pass the black listed libraries from the user. I think the way in address sanitizer looks good.
```sh
$ cat foo.c
#include <stdlib.h>
void bad_foo() {
  int *a = (int*)malloc(40);
  a[10] = 1;
}
int main() { bad_foo(); }

$ cat blacklist.txt
# Ignore reports from bad_foo function.
fun:bad_foo

$ clang -fsanitize=address foo.c ; ./a.out
# AddressSanitizer prints an error report.

$ clang -fsanitize=address -fsanitize-blacklist=blacklist.txt foo.c ; ./a.out
# No error report here.
```

Reply:
Looks reasonable.